packages/aws-cdk-lib/aws-elasticache/lib/elasticache-cluster.ts

```diff
@@ -0,0 +1,139 @@
+import * as cdk from 'aws-cdk-lib';
+import * as elasticache from 'aws-cdk-lib/aws-elasticache';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { Construct } from 'constructs';
+
+export interface ElastiCacheClusterProps {
+  /**
+   * The VPC where the ElastiCache cluster will be deployed
+   */
+  readonly vpc: ec2.IVpc;
+  
+  /**
+   * The cache engine version to use
+   * @default - 6.x (Redis)
+   */
+  readonly engineVersion?: string;
+  
+  /**
+   * The node type for the ElastiCache cluster
+   * @default - cache.t3.micro
+   */
+  readonly nodeType?: string;
+  
+  /**
+   * Number of cache nodes in the cluster
+   * @default 1
+   */
+  readonly numCacheNodes?: number;
+  
+  /**
+   * The name of the cache parameter group to associate with this cache cluster
+   */
+  readonly parameterGroupFamily?: string;
+  
+  /**
+   * Security groups to attach to the cluster
+   * @default - A new security group is created
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+  
+  /**
+   * The weekly time range during which maintenance updates can occur.
+   * @default - No preference
+   */
+  readonly preferredMaintenanceWindow?: string;
+  
+  /**
+   * Specifies the weekly time range during which maintenance 
+   * on the cache cluster is performed
+   */
+  readonly port?: number;
+}
+
+export class ElastiCacheCluster extends Construct {
```

Comment:
this construct is missing the construct interface. You can find more details [here](https://github.com/aws/aws-cdk/blob/d80e3cd8d75b0bb926c6efc1883074a38a8f630d/docs/DESIGN_GUIDELINES.md#construct-interface)

Reply:
This construct also some of the important patterns we usually have in CDK:
 - Imports, which means to have static methods that customer can use them to create an object from this construct for a resource that already exist in their account. Check [here](https://github.com/aws/aws-cdk/blob/d80e3cd8d75b0bb926c6efc1883074a38a8f630d/docs/DESIGN_GUIDELINES.md#imports) from more details
 - Grants, which are the methods that can grant specific role or use to do some actions on the ElastiCacheCluster. Check [here](https://github.com/aws/aws-cdk/blob/d80e3cd8d75b0bb926c6efc1883074a38a8f630d/docs/DESIGN_GUIDELINES.md#grants) for more details
 - Since this construct use the EC2 security groups, so it should implement Ec2 Connections API. Check [here](https://github.com/aws/aws-cdk/blob/d80e3cd8d75b0bb926c6efc1883074a38a8f630d/docs/DESIGN_GUIDELINES.md#connections) for details